### ISSUE TITLE
The result of getpos called in CmdlineLeave is not appropriate

### ISSUE THREAD

**Issue by rapan931** [neovim/neovim#19519]
### Neovim version (nvim -v)

NVIM v0.8.0-dev-nightly-471-gbb7853a62

### Vim (not Nvim) behaves the same?

no, VIM - Vi IMproved 9.0 (2022 Jun 28, compiled Jul 26 2022 00:06:14) Included patches: 1-66

### Operating system/version

Ubuntu 20.04.4(Windows 10 & WSL)

### Terminal name/version

wezterm 20220408-101518-b908e2dd

### $TERM environment variable

xterm-256color

### Installation

build from repo

### How to reproduce the issue

```vim
" minimal.vimrc
augroup test
  autocmd!
  autocmd CmdlineLeave / echomsg getpos('.')
augroup END
```

nvim --clean minimal.vimrc
:source minimal.vimrc<CR>
/END<Esc>
:messages<CR>

### Expected behavior

```
[0, 1, 1, 0] (This is vim behavior)
```
or
```
[0, 4, 9, 0] (If it's the first word of the next hit, I can understand it)
```

### Actual behavior

[0, 4, 12, 0]

**Comment by zeertzjq** [comment]
Yes this is the same problem as #19500, but in this case there is no easy workaround.
